Flaky Jetpack E2E: revise `Advertising: Promote` spec to use the first post and to not create posts unnecessarily.

worldomonation opened the pull request:
Related to: https://github.com/Automattic/wp-calypso/issues/80730, https://github.com/Automattic/wp-calypso/pull/81550.

## Proposed Changes

This PR further revises the Advertising: Promote spec to stop the creation of unnecessary posts.
Furthermore, this PR also changes the selection method of the post from title-based to row number based,

## Testing Instructions

Ensure the following build configurations are passing:
  - [ ] Calypso E2E (desktop)
  - [ ] Jetpack E2E Simple (mobile)
  - [ ] Jetpack E2E Simple (desktop)
  - [ ] Jetpack E2E AT (desktop)

This particular PR survived 50 rounds of repeated testing:

![image](https://github.com/Automattic/wp-calypso/assets/6549265/3e134e57-1358-4a20-b83b-7b3084ef423d)


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

worldomonation committed:
packages/calypso-e2e/src/rest-api-client.ts
- implement new method to see if there are posts on the site.

test/e2e/specs/tools/advertising__promote.ts
- stop creating new post unnecessarily

karenroldan commented on packages/calypso-e2e/src/rest-api-client.ts:
```diff
@@ -635,6 +636,33 @@ export class RestAPIClient {
+	 * @param param1 Keyed object parameter.
+	 * @param {SitePostState} param1.state State of the published post.
+	 */
+	async siteHasPost(
```
**suggestion (non-blocking):** We can also check for the expected response structure & API-specific errors. 

```
if ( response.hasOwnProperty( 'error' ) ) {
	throw new Error(
		`${ ( response as ErrorResponse ).error }: ${ ( response as ErrorResponse ).message }`
	);
}

if (!response || !response.counts || !response.counts.all) {
	throw new Error('Unexpected response structure');
}
```

worldomonation replied:
So, I played around with this a bit, and it appears this endpoint doesn't ever return an error - it only returns a 404 (if _anything_ in the path is wrong, for example site ID) or the actual values.

karenroldan commented on packages/calypso-e2e/src/rest-api-client.ts:
```diff
@@ -635,6 +636,33 @@ export class RestAPIClient {
+			params
+		);
+
+		return response.counts.all[ state ] !== 0 ? true : false;
```
**suggestion (non-blocking):** We can simplify this to `return response.counts.all[state] !== 0` since it already returns a boolean value.

worldomonation replied:
Haha, what an embarrassingly unnecessary ternary.

karenroldan replied:
We've all been there! 😄

karenroldan reviewed (approved):
LGTM! 🚀 left a couple of non-blocking suggestions. Feel free to merge! 👍🏻

worldomonation committed:
packages/calypso-e2e/src/rest-api-client.ts
- remove unnecessary ternary

packages/calypso-e2e/src/types/rest-api-client.types.ts
- add new type to type check the response from PostCounts endpoint.
